pkg/api/v1/registry_test.go

```diff
@@ -0,0 +1,175 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/stacklok/toolhive/pkg/config"
+	"github.com/stacklok/toolhive/pkg/logger"
+)
+
+func MockConfig(t *testing.T, cfg *config.Config) func() {
+	t.Helper()
+
+	// Create a temporary directory for the test
+	tempDir := t.TempDir()
+
+	// TODO: see if there's a way to avoid changing env vars during tests.
+	// Save original XDG_CONFIG_HOME
+	originalXDGConfigHome := os.Getenv("XDG_CONFIG_HOME")
+	t.Setenv("XDG_CONFIG_HOME", tempDir)
+
```

Comment:
This is probably a problem everywhere in our codebase (I'll look into it later), but you need to call `xdg.Reload()` after setting the env var to have it pick up the new value.